chain-impl-mockchain/src/leadership/genesis/mod.rs

```diff
@@ -295,6 +295,49 @@ mod tests {
         );
     }
 
+    #[test]
+    pub fn test_leader_with_invalid_pool_id() {
+        // Arrange
+        let leader_election_parameters = LeaderElectionParameters::new();
+
+        let cb = ConfigBuilder::new()
+            .with_slots_per_epoch(leader_election_parameters.slots_per_epoch)
+            .with_active_slots_coeff(leader_election_parameters.active_slots_coeff_as_milli());
+
+        let mut ledger = LedgerBuilder::from_config(cb)
+            .build()
+            .expect("cannot build test ledger")
+            .ledger;
+
+        let mut pools = HashMap::<PoolId, (SecretKey<RistrettoGroup2HashDh>, u64, Stake)>::new();
+
+        for _i in 0..leader_election_parameters.pools_count {
+            let (pool_id, pool_vrf_private_key) = make_pool(&mut ledger);
+            pools.insert(
+                pool_id.clone(),
+                (pool_vrf_private_key, 0, leader_election_parameters.value),
+            );
+        }
+
+        let date = ledger.date();
```

Comment:
this is only my opinion but i think date variable should be daclared as late as possible. 
 https://stackoverflow.com/questions/10204828/declare-local-variables-as-late-as-possible-or-at-the-nearest-curly-brace-they-b